### PR TITLE
Move workflow update polling inside of interceptor

### PR DESCRIPTION
**Pull request by Sushisource**
## What was changed
Moved update polling to be inside of the interceptor a-la other SDKs

## Why?
Consistent, allows interceptor to change things post-polling

## Checklist

1. Closes https://github.com/temporalio/sdk-java/issues/2094

2. How was this tested:
Added an integ test using the interceptor

3. Any docs updates needed?

### CODE REVIEW

File: temporal-sdk/src/test/java/io/temporal/workflow/updateTest/UpdateTest.java [hunk 1]
```diff
@@ -103,6 +107,55 @@ public void testUpdate() {
...
   }
 
+  private static class FakesResultUpdateInterceptor extends WorkflowClientInterceptorBase {
+    @Override
+    public WorkflowClientCallsInterceptor workflowClientCallsInterceptor(
+        WorkflowClientCallsInterceptor next) {
+      return new WorkflowClientCallsInterceptorBase(next) {
+        @Override
+        public <R> UpdateHandle<R> startUpdate(StartUpdateInput<R> input) {
+          super.startUpdate(input);
```
Comment (Sushisource) [inline temporal-sdk/src/test/java/io/temporal/workflow/updateTest/UpdateTest.java:117]: Side note: Why don't we have something like `InputType.fromExisting(input)` and then some way to relatively easily modify the input? It's a bit annoying to have to re-pass every field through to a new constructor if you want to modify inputs.

I get why they're not directly mutable, but, might be nice to have a convenient way to do that.

File: temporal-opentracing/src/main/java/io/temporal/opentracing/internal/OpenTracingWorkflowClientCallsInterceptor.java [hunk 2]
```diff
@@ -119,7 +120,7 @@ public <R> QueryOutput<R> query(QueryInput<R> input) {
   }
 
   @Override
-  public <R> StartUpdateOutput<R> startUpdate(StartUpdateInput<R> input) {
+  public <R> UpdateHandle<R> startUpdate(StartUpdateInput<R> input) {
```
Comment (Quinn-With-Two-Ns) [inline temporal-opentracing/src/main/java/io/temporal/opentracing/internal/OpenTracingWorkflowClientCallsInterceptor.java:123]: Hm I wonder if we would ever need to return more then `UpdateHandle` here? I think other SDKs just return `UpdateHandle` so if we do will have to address in all SDKs

**Review by Quinn-With-Two-Ns — APPROVED** [review]
LGTM, thanks!

File: temporal-sdk/src/test/java/io/temporal/workflow/updateTest/UpdateTest.java (hunk 1 above)
Comment (Quinn-With-Two-Ns) [reply temporal-sdk/src/test/java/io/temporal/workflow/updateTest/UpdateTest.java:117]: Modify inputs is not really an encouraged pattern. Not that some users don't but I it is a advanced path to take.

**Pull request merged**